packages/api/amplify_api/example/integration_test/util.dart

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:async';
+
+import 'package:amplify_api/amplify_api.dart';
+import 'package:amplify_api_example/amplifyconfiguration.dart';
+import 'package:amplify_api_example/models/ModelProvider.dart';
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:amplify_flutter/amplify_flutter.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+const _subscriptionTimeoutInterval = 5;
+
+String? testUsername;
```

Comment:
I guess this is fine as long as you are only using one user per test. I might still put this in a class so that you don't run into some hard to track down global variable issue later.

```dart
class TestUser {
  TestUser({
    String? username,
    String? password,
  })  : _username = 'testUser${uuid()}',
        _password = uuid(secure: true);

  final String _username;
  final String _password;

  Future<void> signUp() async {
    await signOut();
    final testEmail = '$_username@amazon.com';
    final result = await Amplify.Auth.signUp(
      username: _username,
      password: _password,
      options: CognitoSignUpOptions(
        userAttributes: {CognitoUserAttributeKey.email: testEmail},
      ),
    );
    if (!result.isSignUpComplete) {
      throw const AmplifyException('Unable to sign up test user.');
    }
  }

  Future<void> signOut() async {
    final session = await Amplify.Auth.fetchAuthSession();
    if (!session.isSignedIn) return;
    await Amplify.Auth.signOut();
  }
}
```


```dart
final user = TestUser();
await user.signIn();
```

Reply:
good suggestion, just added something w a few mods